buffer-api/src/main/java/io/netty/buffer/api/BufferHolder.java

```diff
@@ -125,27 +96,9 @@ protected final void replaceBuf(Buffer newBuf) {
      *
      * @param send The new {@link Buffer} instance that is replacing the currently held buffer.
      */
-    protected final void replaceBuf(Send<Buffer> send) {
-        try (var ignore = buf) {
-            buf = send.receive();
-        }
-    }
-
-    /**
-     * Replace the underlying referenced buffer with the given buffer.
-     * <p>
-     * This method is protected to permit advanced use cases of {@link BufferHolder} sub-class implementations.
-     * <p>
-     * <strong>Note:</strong> this method decreases the reference count of the current buffer,
-     * and increases the reference count of the new buffer.
-     * <p>
-     * The buffer assignment is performed using a volatile store.
-     *
-     * @param newBuf The new {@link Buffer} instance that is replacing the currently held buffer.
-     */
-    protected final void replaceBufVolatile(Buffer newBuf) {
-        var prev = (Buffer) BUF.getAndSet(this, newBuf.acquire());
-        prev.close();
+    protected final void replaceBuffer(Send<Buffer> send) {
+        buf.close();
+        buf = send.receive();
```

Comment:
Could the `receive()` throw ? It might be safer to only call `close` once we know we can receive a new buffer otherwise we might end up in a strange state